addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -4,52 +4,81 @@ import "@website/snippets/s_website_form/000";  // force deps
+        if (this.$('.s_turnstile').length) {
+            this.$('.s_turnstile').remove();
+        }
+    },
+
+    /**
+     * @override
+     * Discard all library changes to reset the state of the Html.
+     */
+    destroy: function () {
+        this.cleanTurnstile();
+        this._super(...arguments);
+    },
+};
+
+const signupTurnStile = {
+    ...turnStile,
+    async willStart() {
+        this._super(...arguments);
+        this.cleanTurnstile();
+        this.addTurnstile(this.action)?.insertBefore(`${this.selector} button[type='submit']`);
+    },
+};
+
 publicWidget.registry.s_website_form.include({
-        /**
-         * @override
-         */
-        start: function () {
-            const res = this._super(...arguments);
-            this.cleanTurnstile();
-            if (!this.isEditable && !this.$('.s_turnstile').length && session.turnstile_site_key) {
-                const mode = new URLSearchParams(window.location.search).get('cf') == 'show' ? 'always' : 'interaction-only';
-                $(`<div class="s_turnstile cf-turnstile float-end"
-                         data-action="website_form"
-                         data-appearance="${mode}"
-                         data-response-field-name="turnstile_captcha"
-                         data-sitekey="${session.turnstile_site_key}"
-                         data-error-callback="throwTurnstileError"
-                    ></div>
-                    <script class="s_turnstile">
-                        // Rethrow the error, or we only will catch a "Script error" without any info 
-                        // because of the script api.js originating from a different domain.
-                        function throwTurnstileError(code) {
-                            const error = new Error("Turnstile Error");
-                            error.code = code;
-                            throw error;
-                        }
-                    </script>
-                    <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
-                `).insertAfter('.s_website_form_send, .o_website_form_send');
-            }
-            return res;
-        },
-
-        /**
-         * Remove potential existing loaded script/token
-         */
-        cleanTurnstile: function () {
-            if (this.$('.s_turnstile').length) {
-                this.$('.s_turnstile').remove();
-            }
-        },
-
-        /**
-         * @override
-         * Discard all library changes to reset the state of the Html.
-         */
-        destroy: function () {
-            this.cleanTurnstile();
-            this._super(...arguments);
-        },
+    ...turnStile,
+
+    /**
+     * @override
+     */
+    start: function () {
+        const res = this._super(...arguments);
+        this.cleanTurnstile();
+        this.addTurnstile("website_form")?.insertAfter('.s_website_form_send, .o_website_form_send');
```

Comment:
As advertised, this is probably bad but in rare cases not concerned by this PR. I guess we can merge this PR as it is and I'll review more with https://github.com/odoo/odoo/pull/188660 later.